parl/framework/agent.py

```diff
@@ -0,0 +1,95 @@
+#   Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import ABCMeta, abstractmethod
+from multiprocessing import Process, Value
+from py_simulator import Simulator
+from parl.common.replay_buffer import Experience
+from parl.common.logging import GameLogEntry
+
+
+class AgentBase(Process):
+    __metaclass__ = ABCMeta
+
+    def __init__(self, game_name, game_options, num_games):
+        super(AgentBase, self).__init__()
+        self.id = -1  # isolated agent, not recognized by the framework yet
```

Comment:
what do you mean by "isolated agent"?

Reply:
It means the agent is just created and has not been added to the Robot yet.

Reply:
rephrase because this meaning is not properly expressed